pusher-platform-core/src/main/kotlin/com/pusher/platform/subscription/RetryingSubscription.kt

```diff
@@ -111,6 +107,7 @@ class RetryingSubscription(
 
         init {
             logger.verbose("${RetryingSubscription@this}: transitioning to RetryingSubscriptionState")
+            listeners.onError(error)
```

Comment:
I'm not sure we want to communicate the error to the listener here. 

The `onError` listener is only supposed to be called at most once for a subscription (at the outermost level, if that makes sense). i.e. If a `BaseSubscription` encounters an error then it'll call its `onError` listener because it doesn't have any retry / resuming behaviour. However, if a `RetryingSubscription` encounters an error (or one bubbles up to it) then we only want to call the `onError` listener if the subscription isn't going to retry.

In this case I think the transition to the `FailedSubscriptionState` is what should eventually lead to the error being communicated to the relevant listener.

Reply:
I can't remember specifics, but the integration tests as they were defined on `platform-js`  and this was needed in order for it to pass. I'll double check thought.

Reply:
Just tried it without that and this use case fails:
https://github.com/pusher/pusher-platform-android/pull/27/files/d19c0b2476f69ccc572eb9d448a6a3e0e59b64d0#diff-857fb2589d7ff72ddb60112b1a61dc3bR105

Which is homologous to this one:
https://github.com/pusher/pusher-platform-js/blob/master/test/integration/subscribe.test.js#L140

Reply:
Interesting - I think the `onError` listener should be being called here: https://github.com/pusher/pusher-platform-android/blob/d19c0b2476f69ccc572eb9d448a6a3e0e59b64d0/pusher-platform-core/src/main/kotlin/com/pusher/platform/subscription/RetryingSubscription.kt#L193-L200

Is it not getting into the failed state?